.github/workflows/testing_changes.yml

```diff
@@ -12,14 +12,45 @@
 #   - cargo tarpaulin does not work on macos
 #   - cargo test fails with linking on macos
 
-on: [push, pull_request]
+on:
+  push:
+    paths-ignore:
+      - '.github/workflows/**'
+  pull_request:
+    types: [opened,reopened,synchronize,ready_for_review,labeled]
 
 name: testing_changes
 
 jobs:
+  run-guard:
+    # other jobs depend on this one using 'needs: run-guard' to avoid running when it is skipped.
+    # it succeeds if any of the following conditions are met:
+    #   - it is not a pr-labeling event AND the pr is not labeled 'prevent-ci' AND it is not a draft pr
+    #   - it is not a pr-labeling event AND the pr is labeled 'force-ci'
+    #   - it is a pr-labeling event AND the label being added is 'force-ci'
+    #
+    # on success it dumps the action's 'github' context
+    runs-on: ubuntu-latest
+    name: 'run-guard'
+    if: |
+      (
+           (github.event.action != 'labeled')
+        && !contains( github.event.pull_request.labels.*.name, 'prevent-ci')
+        && !github.event.pull_request.draft
+      )
+      || ( (github.event.action != 'labeled') && contains( github.event.pull_request.labels.*.name, 'force-ci' ) )
+      || (github.event.label.name == 'force-ci')
+    steps:
+        #helpful for debugging event
+      - name: dump action context 'github'
+        env:
+          GITHUB_CONTEXT: ${{ toJson(github) }}
+        run: echo "$GITHUB_CONTEXT"
```

Comment:
This `echo` can leak any restricted information or s3cre3t?

Reply:
You can take a look at it [here](https://github.com/fernandobatels/rsfbclient/runs/1266171580?check_suite_focus=true#step:2:589).
I did not find anything concerning, and think it cannot contain any restricted information as it is a public repository anyways.

Reply:
github is supposed to redact this information automatically you can see this in the `token: "***"` near the beginning of the output for that `echo`

but now that you mention it I can't really find enough information to be 100% sure that this will never be a problem
so I changed it